cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf

```diff
@@ -0,0 +1,51 @@
+#include "..\script_component.hpp";
+/*
+ * Author: SGT.Sweetwater.I
+ * This module function set damage to all lights indicated in a array within a selected radius from module position.
+ * Gives the effect of a Power Outage.
+ * 
+ * Arguments:
+ * 0: None
+ *
+ * Example:
+ * this call cScripts_fnc_modulePowerOutage
+ *
+ * Public: No
+ */
+
+private _center = position _logic;
+private _OutageArray = [];
+private _OutageRadius = [""];
+
+    _dialogResult = [
+    "Power Outage",
+    [
+        ["Meters","", 1200]
+    ]
+] call Ares_fnc_showChooseDialog;
+  if (count _dialogResult == 0) exitWith {};
+
+  private _OutageRadius = _dialogResult select 0;
+  private _OutageRadius = parseNumber _OutageRadius;
+  Private _OutageArray = [
```

Comment:
Possible add all cup lamps as well?

Reply:
I'll check through Cups terrains / ..mod and see if I can find a class list for those to add into the array. Also from further testing it appears Fuel stations are exempt. still working out what is needed for those as well.